README.md

```diff
@@ -8,16 +8,207 @@ Local.
+
+### Conflict Control
+
+TBD. 
+
+The basic idea is when modifying a document, user needs to control conflict by handling the revision of a document, currently the connector controls this process, after retriving the latest revision, connector uses it to update/delete doc, and returns 409 conflict error if doc changes during that time slot. In the middle, user could not interfere and provide their own conflict solution.
+
+This part will be done after we complete the `_rev` property refactor.
+
+## Model 
+
+### Map between model and document
+
+Unlike relational db or mongodb, Cloudant doesn't have a concept as 'table' or 'collection', data in a Cloudant database are all stored as documents.
+
+The connector uses a design document to represent a Loopback model, and common documents to represent model instances.
+
+The following is a comparison among different databases:
+
+|               | Model                        | Model Property                        | Model Instance         |
+|---------------|------------------------------|---------------------------------------|------------------------|
+| Relational DB | table                        | column in table                       | row in table           |
+| Mongodb       | collection                   | createIndex if property.index is true | document in collection |
+| Cloudant      | Design documents in database | NOT stored in document                     | common documents in database  |
+
+To create a model, the connector creates a design document with the following config:
+
+```
+type: 'text',
+name: 'lb-index-' + modelName,
+ddoc: 'lb-index-ddoc-' + modelName,
+index: {
+  default_field: {
+    enabled: false,
+  },
+  selector: {
+    [modelIndex]: modelName
+  },
+},
+```
+
+By default, `modelIndex` is 'loopback__model__name', and `modelSelector` is {[modelIndex]: modelName}. User can customize `modelSelector` and `modelIndex` in datasource's json file, for details please check [model-specific configuration](https://github.com/strongloop/loopback-connector-cloudant##model-specific-configuration)
+
+To create a model instance, the connector creates a non-design document with value of property 'loopback__model__name' equals to `modelName`. 
 
-* Uses Cloudant Query (Lucene) to support ad-hoc searching.
-* [Loopback query](http://loopback.io/doc/en/lb3/Querying-data.html) support for: fields, limit, order, skip and where filters.
-* Query and filtering is performed on the database for optimal efficiency.
-* Use different DB instances per model definition.
-* Supports basic model discovery.
+For model properties, we plan to create index for property that has config `index: true`. In the future, it will be the same way as what mongodb connector does.
```

Comment:
can you also add a statement to say what happens when `index: false` or not specified?